app/javascript/src/ActiveDocs/ThreeScaleApiDocs.ts

```diff
@@ -17,6 +20,96 @@ const appendSwaggerDiv = (container: HTMLElement, id: string): void => {
   container.appendChild(div)
 }
 
+/**
+ * when using Record notation, the following error is thrown:
+ * 'TS2456: Type alias 'BodyValue' circularly references itself.'
+ */
+// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
+type BodyValue = boolean | number | string | { [key: string]: BodyValue }
```

Comment:
I know not much of TS, I understand this means:

`BodyValue` is either a `boolean`, a `number`, a `string`, or an object with only  one item which key is a `string` and the value is a `BodyValue`.

Am I right?  I'm confused by the `[key: string]` syntax, is `key` a language keyword here?

Reply:
I don't know TS either :grimacing: 

Your understanding is almost correct, except that `{ [key: string]: BodyValue }` means that BodyValue can be an object with **any** number (not only one) of items whose key is a `string` and the value is `BodyValue`. This is one way to define an object in TS, see https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures
Another would be `Record<string, BodyValue>` (see https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type), but this was giving me an error as explained in the comments on lines 24-25.

Reply:
I understand, so the `[]` mean and `Array`, thanks!